support-frontend/tsconfig.json

```diff
@@ -20,7 +20,8 @@
 		// These override @guardian/tsconfig as they have introduced a lot of errors.
 		// The number of errors indicated below is at the time of writing to give an idea of the scale of the fix needed.
 		// We should look at removing these as soon as we can.
-		"noUncheckedIndexedAccess": false // 209 errors in 63 files.
+		"noUncheckedIndexedAccess": false, // 209 errors in 63 files.
+		"types": ["@testing-library/jest-dom"]
```

Comment:
Without this I was getting a large number of errors like this from the TypeScript compiler:

```
error TS2339: Property 'toBeInTheDocument' does not exist on type 'Matchers<void, HTMLElement | null>'.

133     expect(screen.queryByText('Direct debit')).not.toBeInTheDocument();
```

Reply:
Which change caused this? Was it adding `@types/jest`?

Reply:
Good question @andrewHEguardian. No it wasn't the addition of `@types/jest`, that was added to fix a whole load of similar type errors, so it was the combination of adding that package and the change on this line (`"types": ["@testing-library/jest-dom"]`) which got TypeScript happy again. I'm not 100% of the cause, only that it happened after upgrading storybook. I'll dig a bit more and see if I can find out.

Reply:
After some digging I think I've gotten to the bottom of this:

## Before

Storybook v7 brought in `@types/testing-library__jest-dom` as a transitive dependency and it was what was providing the types for `.toBeInTheDocument` & friends. That package is now [deprecated](https://www.npmjs.com/package/@types/testing-library__jest-dom) as `@testing-library/js-dom` now includes types. We were getting the types for free as packages under `@types` are [included by default](https://www.typescriptlang.org/tsconfig/#typeRoots) (TIL!)

## After

After upgrading to Storybook v8 we've lost `@types/testing-library__jest-dom` from the bundle and should be getting the types from `@testing-library/js-dom`. But they aren't included for free (as they aren't namespaced under `@types`) so we need to include them by adding this line to our tsconfig.